File: leetcode.py/除自身以外数组的乘积.py
class Solution(object):
    def productExceptSelf(self, nums):
        n = len(nums)
        answer = [1] * n
        prefix = 1
        for i in range(n):
            answer[i] = prefix
            prefix *= nums[i]
        
        suffix = 1
        for i in range(n-1,-1,-1):
            answer[i] *= suffix
            suffix*= nums[i]

        return answer
# 以下是代码使用示例
if __name__ == "__main__":
    solution = Solution()
    nums = [1, 2, 3, 4]
    result = solution.productExceptSelf(nums)
    print(f"输入数组: {nums}")
    print(f"除自身以外数组的乘积: {result}")

File: leetcode.js/除自身以外数组的乘积.js
var productExceptSelf = function(nums) {
    const n = nums.length;
    const answer = Array(n).fill(1);
    let prefix = 1;
    for(let i=0 ; i<n ; i++){
        answer[i] = prefix;
        prefix *= nums[i];
    }
    let suffix = 1
    for(let i = n-1 ; i >-1 ; i--){
        answer[i] *= suffix;
        suffix *= nums[i];
    }
    return answer;
};
// 示例用法
const nums = [1, 2, 3, 4];
const result = productExceptSelf(nums);
console.log(result); // 输出: [24, 12, 8, 6]
